### PR TITLE
solution added

### PR DESCRIPTION
**Pull request by IrynaDim**

### CODE REVIEW

File: src/main/webapp/WEB-INF/views/products/all.jsp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            </td>
+            <td>
+                <a href="${pageContext.request.contextPath}/shopping-cart/products/add?id=${product.id}">Buy</a>
+            </td>
+        </tr>
+    </c:forEach>
+</table>
+<a href="${pageContext.request.contextPath}/shopping-cart/products">Go to shopping cart</a>
+</body>
+</html>
```
Comment (DmGrt) [inline src/main/webapp/WEB-INF/views/products/all.jsp]: add an empty line)

**Review by DmGrt — APPROVED** [review]
Well done! just one empty line needed:)

File: src/main/java/com/internet/shop/controllers/AddToCartController.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.ProductService;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddToCartController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/AddToCartController.java]:
```suggestion
public class AddProductToCartController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/DeleteUsersController.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+    private ShoppingCartService shoppingCartService
+            = (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String userId = req.getParameter("id");
+        Long id = Long.valueOf(userId);
+        userService.delete(id);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/DeleteUsersController.java:24]: let's delete shopping cart here as well

File: src/main/java/com/internet/shop/controllers/ProductController.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ProductController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/ProductController.java]:
```suggestion
public class GetAllProductsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/ShoppingCartController.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ShoppingCartController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/ShoppingCartController.java]:
```suggestion
public class GetCartProductsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/UserController.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import com.internet.shop.model.User;
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class UserController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/UserController.java]:
```suggestion
public class GetAllUsersController extends HttpServlet {
```

File: src/main/webapp/WEB-INF/views/products/all.jsp (hunk 1 above)
Comment (RomanDubovskyi) [reply src/main/webapp/WEB-INF/views/products/all.jsp]: you still need to add empty line

File: src/main/webapp/web.xml [hunk 7]
```diff
@@ -7,8 +7,90 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addProduct</servlet-name>
+        <servlet-class>com.internet.shop.controllers.AddProductController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>addProduct</servlet-name>
+        <url-pattern>/products/addProduct</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/products/add</url-pattern>
```

File: src/main/webapp/WEB-INF/views/products/all.jsp (hunk 1 above)
Comment (IrynaDim) [reply src/main/webapp/WEB-INF/views/products/all.jsp]: done

File: src/main/java/com/internet/shop/controllers/AddToCartController.java (hunk 2 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/controllers/AddToCartController.java]: done

File: src/main/java/com/internet/shop/controllers/ProductController.java (hunk 4 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/controllers/ProductController.java]: done

File: src/main/java/com/internet/shop/controllers/ShoppingCartController.java (hunk 5 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/controllers/ShoppingCartController.java]: done

File: src/main/java/com/internet/shop/controllers/UserController.java (hunk 6 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/controllers/UserController.java]: done

File: src/main/webapp/web.xml (hunk 7 above)
Comment (IrynaDim) [reply src/main/webapp/web.xml]: done

File: src/main/java/com/internet/shop/controllers/InjectDataController.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class InjectDataController extends HttpServlet {
+    public static final Injector injector = Injector.getInstance("com.internet.shop");
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+    private final ShoppingCartService shoppingCart = (ShoppingCartService) injector
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/InjectDataController.java]: why this service is final and others are not? 
keep your code consistent: make all of them final, or non final

File: src/main/java/com/internet/shop/controllers/AddToCartController.java (hunk 2 above)
Comment (RomanDubovskyi) [reply src/main/java/com/internet/shop/controllers/AddToCartController.java]: it's still not done

File: src/main/java/com/internet/shop/controllers/AddProductToCartController.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddProductToCartController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/AddProductToCartController.java]:
```suggestion
public class AddProductController extends HttpServlet {
```

**Review by RomanDubovskyi — APPROVED** [review]
altogether it's good, just fix last comments before merging PR

**Pull request merged**